factory-package-news/factory-package-news-web.py

```diff
@@ -26,16 +26,21 @@
 import re
 import os
 import sys
+from urlparse import urlparse
 
 import re
 digits_re = re.compile('^[0-9]+$')
 
-_dir = "/var/lib/openqa/factory/snapshot-changes"
+BASE_DIR = '/var/lib'
 
 app = Flask(__name__)
 
+def get_dir(url):
+    return os.path.join(BASE_DIR, urlparse(url).path.lstrip('/'))
```

Comment:
this needs some sanitation to prevent directory traversal attacks. Put e.g. a regexp to make sure it's the expected path.